src/KilnUniV2LPSwap.sol

```diff
@@ -56,42 +56,109 @@ interface UniswapRouterV2Like {
     ) external returns (uint amountA, uint amountB);
 }
 
+interface UniswapPairV2Like {
+    function getReserves() external view returns (uint112, uint112, uint32);
+}
+
+interface PipLike {
+    function peek() external returns (bytes32, bool);
+}
+
 contract KilnUniV2LPSwap is KilnBase {
 
+    event Debug(uint256, uint256);
+
     using UniswapV2Library for *;
 
     address public immutable uniV2Router;
     address public immutable pairToken;
     address public immutable receiver;
 
-    event LogI(uint256);
+    uint256 internal constant WAD = 10 ** 18;
+
+    uint256 public max;  // [WAD] Maximum acceptable buy price in sell.
+    uint256 public nip;  // [WAD] Maximum percentage of pool sell reserves to utilize.
+    address public pip;  // (Optional) pricing module
 
     constructor(address _sell, address _buy, address _uniV2Router, address _receiver) KilnBase(_sell, _buy) {
         receiver = _receiver;
         uniV2Router = _uniV2Router;
+        nip = WAD / 100;  // Default to 1% of available reserve
         pairToken = UniswapRouterV2Like(_uniV2Router).factory().pairFor(_sell, _buy);
     }
 
+    //rounds to zero if x*y < WAD / 2
+    function _wmul(uint x, uint y) internal pure returns (uint z) {
+        z = (x * y) + (WAD / 2) / WAD;
+    }
+
+    //rounds to zero if x*y < WAD / 2
+    function _wdiv(uint x, uint y) internal pure returns (uint z) {
+        z = (x * WAD) + (y / 2) / y;
+    }
```

Comment:
Did they change the way order of operations works in Solidity, or do these need some parentheses?

```diff
    //rounds to zero if x*y < WAD / 2
    function _wmul(uint x, uint y) internal pure returns (uint z) {
-        z = (x * y) + (WAD / 2) / WAD;
+        z = ( (x * y) + (WAD / 2) ) / WAD;
    }

    //rounds to zero if x*y < WAD / 2
    function _wdiv(uint x, uint y) internal pure returns (uint z) {
-        z = (x * WAD) + (y / 2) / y;
+        z = ( (x * WAD) + (y / 2) ) / y;
    }
```

Also comment on the division function is wrong, copy-pasted from the multiplication function.

Reply:
Bless your heart. I'd fixed this locally and hadn't pushed. I'm going to revert the nip stuff from this commit because it breaks things and I think it's overengineering at this point, but I'll make sure this propagates.